Exclude _testclinic_depr.c.h from c-analyzer

zware opened the pull request:
Also revert `_testclinic.c` to match 3.14 as a test to make sure this will fix CI for the 3.14 branch.

zware committed:
Exclude _testclinic_depr.c.h from c-analyzer

_testclinic.c mocks out PY_VERSION_HEX to 3.8 before including
_testclinic_depr.c.h to avoid the errors the preprocessor would
otherwise throw due to the deprecation feature it is testing.

zware committed:
Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

This restores Modules/_testclinic.c to match the same file in the 3.14
branch.

zware committed:
amend! Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

This restores Modules/_testclinic.c to match the same file in the 3.14
branch.

emmatyping reviewed (approved):
I can confirm this passes `make check-c-globals` locally. Thanks for fixing this!

zware committed:
amend! amend! Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

This restores Modules/_testclinic.c to match the same file in the 3.14
branch.

zware committed:
amend! amend! amend! Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

Partially revert 74e2acddf68b31ce16e8e0067b1df8c7b67bd6c8

This restores Modules/_testclinic.c to match the same file in the 3.14
branch.
